lib/datastore/transaction.js

```diff
@@ -195,14 +199,29 @@ Transaction.prototype.commit = function(callback) {
 
   this.request_(protoOpts, reqOpts, function(err, resp) {
     if (err) {
-      callback(err, resp);
+      // Rollback automatically for the user.
+      self.rollback(function(err) {
+        if (err) {
+          err.message = [
```

Comment:
I think we can just eat this error -- right now rollbacks are fine as best effort.